Model/Api/GetProduct.php

```diff
@@ -165,6 +174,22 @@ private function getProductFamily(){
                 array_push($childrenStockArray, $childProductInventory);
             }
             $this->productData->setChildren($childrenStockArray);
+
+            $attributes = $parentProduct->getTypeInstance()->getConfigurableOptions($parentProduct);
+            $options = array();
+
+            foreach($attributes as $attr) {
+                foreach ($attr as $p) {
+                    $attribute = $this->eavConfig->getAttribute('catalog_product', $p['attribute_code']);
+                    if ($this->productData->getOptions()) {
+                        $update = array_merge($this->productData->getOptions(), $attribute->getSource()->getAllOptions());
+                        $this->productData->setOptions($update);
+                    } else {
+                        $this->productData->setOptions($attribute->getSource()->getAllOptions());
+                    }
+                }
+            }
```

Comment:
This may probably produce many duplicate option entries, as children of the configurable products have the same options as parents.

Reply:
I am only running this on the parent product see line 178. Then I am looping through all the attributes of that one parent product. 

Do I need to worry about attributes sharing the same options? Could I easily add this to a set then change to an array?